app/uk/gov/hmrc/entrydeclarationdecision/services/ProcessDecisionService.scala

```diff
@@ -56,33 +57,39 @@ class ProcessDecisionService @Inject()(
 
       val amendment = isAmendment(decision.metadata.messageType)
 
-      decision.response match {
-        case resp: DecisionResponse.Acceptance =>
-          if (amendment) {
-            doProcessDecision(
-              decision.withResponse(resp),
-              amendmentAcceptanceXMLBuilder,
-              storeConnector.getAcceptanceEnrichment(_, amendment))
-          } else {
-            doProcessDecision(
-              decision.withResponse(resp),
-              declarationAcceptanceXMLBuilder,
-              storeConnector.getAcceptanceEnrichment(_, amendment))
-          }
-
-        case resp: DecisionResponse.Rejection =>
-          if (amendment) {
-            doProcessDecision(
-              decision.withResponse(resp),
-              amendmentRejectionXMLBuilder,
-              storeConnector.getAmendmentRejectionEnrichment)
-          } else {
-            doProcessDecision(
-              decision.withResponse(resp),
-              declarationRejectionXMLBuilder,
-              _ => Future.successful(Right(DeclarationRejectionEnrichment)))
-          }
+      def matchDecisionResponse(decisionResponse: DecisionResponse) =
```

Comment:
```suggestion
      def processDecisionResponse(decisionResponse: DecisionResponse) =
```
The fact that is is implemented as a 'match' is coincidental and hides the meaning of what it is doing.